### PR TITLE
Mini changes pulled from #2146

### PR DESCRIPTION
**Pull request by coolreader18**

### CODE REVIEW

File: vm/src/stdlib/collections.rs [hunk 1]
```diff
@@ -247,7 +247,32 @@ mod _collections {
             self.maxlen.store(maxlen);
         }
 
-        #[pymethod(name = "__repr__")]
+        #[pymethod(magic)]
+        fn getitem(&self, idx: isize, vm: &VirtualMachine) -> PyResult {
```
Comment (youknowone) [inline vm/src/stdlib/collections.rs:251]: oh.. it even didn't have getitem

File: vm/src/obj/objint.rs [hunk 2]
```diff
@@ -92,20 +93,24 @@ macro_rules! impl_into_pyobject_int {
 
 impl_into_pyobject_int!(isize i8 i16 i32 i64 usize u8 u16 u32 u64 BigInt);
 
+pub fn try_to_primitive<'a, I>(i: &'a BigInt, vm: &VirtualMachine) -> PyResult<I>
```
Comment (youknowone) [inline vm/src/obj/objint.rs:96]: 👍

File: vm/src/cformat.rs [hunk 3]
```diff
@@ -420,25 +435,22 @@ impl CFormatString {
...
-                                // TODO: handle errors when truncating BigInt to usize
-                                *q = Some(CFormatQuantity::Amount(
-                                    objint::get_value(&width_obj).to_usize().unwrap(),
-                                ));
-                                Ok(tuple_index)
-                            }
+                Some(CFormatQuantity::FromValuesTuple) => match elements.next() {
+                    Some(width_obj) => {
+                        tuple_index += 1;
+                        if !objtype::isinstance(&width_obj, &vm.ctx.types.int_type) {
```
Comment (youknowone) [inline vm/src/cformat.rs:441]: swapping if and else would make else block a lot cleaner.
but it seems it is not part of this PR

**Review by youknowone — APPROVED** [review]

File: vm/src/stdlib/collections.rs (hunk 1 above)
Comment (coolreader18) [reply vm/src/stdlib/collections.rs:251]: Yeah, I had the same reaction 😁
